source/dofs/dof_tools.cc

```diff
@@ -765,6 +765,140 @@ namespace DoFTools
 
 
 
+  template <typename DoFHandlerType>
+  IndexSet
+  extract_dofs_with_support_on(const DoFHandlerType &dof_handler,
+                               const std_cxx11::function< bool(const typename DoFHandlerType::active_cell_iterator &)> &predicate,
+                               const ConstraintMatrix &cm)
+  {
+    const std_cxx11::function< bool(const typename DoFHandlerType::active_cell_iterator &)> predicate_local
+    = [=] (const typename DoFHandlerType::active_cell_iterator & cell) -> bool
+    {
+      return cell->is_locally_owned() && predicate(cell);
+    };
```

Comment:
you need 5 lines of code here where you could have just put the same condition into your `if` statement below :-)